File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/Termica/TermicaCommand.cs
using Newtonsoft.Json;

namespace Xamarin_Android_Intent_Digital_Hub.IntentServices.Termica
{
    /**
     * Classe que generaliza todos os comandos do módulo TERMICA, definindo o módulo do comando e o seu tipo de retorno
     */
    abstract class TermicaCommand : IntentDigitalHubCommand
    {
        //O retorno dos comandos de impressora é sempre um int
        [JsonProperty(PropertyName = "resultado")]
        private readonly int resultado;

        protected TermicaCommand(string functionName) : base(functionName, IntentDigitalHubModule.TERMICA) { }

        public int GetResultado()
        {
            return resultado;
        }
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/TERMICA/TermicaCommand.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.IntentDigitalHubCommand
import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.IntentDigitalHubModule
import com.google.gson.annotations.SerializedName

/**
 * Classe que generaliza todos os comandos do módulo TERMICA, definindo o módulo do comando e o seu tipo de retorno
 */

abstract class TermicaCommand protected constructor(functionName: String) :
    IntentDigitalHubCommand(functionName, IntentDigitalHubModule.TERMICA) {
    /**
     * O retorno dos comandos TERMICA é sempre um inteiro, as classes de comando são utilizadas, também, para deserializar os JSON de RETORNO
     * do IDH, portanto é definido um objeto para que seja possível serializar o retorno de um comando dentro de um objeto da classe correspondente
     * a esse comando; verifique a impĺementação de @onActivityResult nos módulos implementados no projeto.
     */

    @SerializedName("resultado")
    val resultado = 0
}